source/retarget.cpp

```diff
@@ -152,7 +157,15 @@ extern "C" FILEHANDLE PREFIX(_open)(const char* name, int openmode) {
         init_serial();
         return 2;
     }
-    #endif
+#ifdef YOTTA_CFG_DEBUG_OPTIONS_COVERAGE
+    else if(coverage_report) {
+        init_serial();
+        notify_coverage_start(name);
+        return 3;
+    }
+#endif
+
+#endif
```

Comment:
It is not necessary at the moment because we don't have any file systems enabled it. When we do, if file handle #3 is open by the application when you do the coverage dumping, the results might be weird. I'm not actually sure that can happen in practice, it just seems the logical thing to do.